From ea01dfcf4c22aaadf94c7b20cbae7ee64b84892a Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Wed, 22 Apr 2015 21:47:57 +0200
Subject: tfm: drop __DATE__

to make it reproducible

Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 libclamav/tomsfastmath/misc/fp_ident.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libclamav/tomsfastmath/misc/fp_ident.c b/libclamav/tomsfastmath/misc/fp_ident.c
index ba9c64c..274a34f 100644
--- a/libclamav/tomsfastmath/misc/fp_ident.c
+++ b/libclamav/tomsfastmath/misc/fp_ident.c
@@ -15,7 +15,7 @@ const char *fp_ident(void)
 
    memset(buf, 0, sizeof(buf));
    snprintf(buf, sizeof(buf)-1,
-"TomsFastMath (%s)\n"
+"TomsFastMath\n"
 "\n"
 "Sizeofs\n"
 "\tfp_digit = %lu\n"
@@ -70,7 +70,7 @@ const char *fp_ident(void)
 #ifdef TFM_HUGE
 " TFM_HUGE "
 #endif
-"\n", __DATE__, sizeof(fp_digit), sizeof(fp_word), FP_MAX_SIZE);
+"\n", sizeof(fp_digit), sizeof(fp_word), FP_MAX_SIZE);
 
    if (sizeof(fp_digit) == sizeof(fp_word)) {
       strncat(buf, "WARNING: sizeof(fp_digit) == sizeof(fp_word), this build is likely to not work properly.\n", 
