From fa38b9113bd0c3d231e1151b995a476374769eff Mon Sep 17 00:00:00 2001
From: Micah Snyder <micasnyd@cisco.com>
Date: Thu, 18 Jan 2018 11:27:39 -0500
Subject: Updating version numbers and adding information about the security
 patch release to the readme.

Patch-Name: Updating-version-numbers-and-adding-information-abou.patch
---
 ChangeLog                    |  4 ++++
 README                       | 25 ++++++++++++++++++++++++-
 configure.ac                 |  2 +-
 docs/clamdoc.tex             |  7 ++++---
 m4/reorganization/version.m4 |  2 +-
 5 files changed, 34 insertions(+), 6 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 337d953..6fa1619 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+Mon, 18 Jan 2018 12:45:00 -0500 (Steven Morgan)
+------------------------------------------
+ * ClamAV 0.99.2.1 security patch release.
+
 Thu, 22 Apr 2016 12:45:00 -0500 (Steven Morgan)
 ------------------------------------------
  * ClamAV 0.99.2 release.
diff --git a/README b/README
index 0059252..059ea90 100644
--- a/README
+++ b/README
@@ -2,6 +2,29 @@ Note: This README/NEWS file refers to the source tarball. Some things described
 here may not be available in binary packages.
 --
 
+0.99.2.1
+------
+
+ClamAV 0.99.2.1 is a hotfix release to patch a set of vulnerabilities.
+
+    - fixes for the following CVE's: CVE-2017-6418, CVE-2017-6420, 
+      CVE-2017-12374, CVE-2017-12375, CVE-2017-12376, CVE-2017-12377, 
+      CVE-2017-12378, CVE-2017-12379, CVE-2017-12380. 
+    - also included are 2 minor fixes to properly detect openssl install
+      locations on FreeBSD 11, and prevent false warnings about zlib 1.2.1#
+      version numbers. 
+
+Thank you to the following ClamAV community members for your code
+submissions and bug reports! 
+
+Alberto Garcia
+Daniel J. Luke
+Francisco Oca
+Sebastian A. Siewior
+Suleman Ali
+
+Special thanks to Offensive Research at Salesforce.com for responsible disclosure.
+
 0.99.2
 ------
 
@@ -2052,7 +2075,7 @@ document and contact our administrator - Luca Gibelli <nervous*clamav.net>.
 -) documentation:
     + new Spanish documentation on ClamAV and Sendmail integration by
       Erick Ivaan Lopez Carreon
-    + included clamdoc.pdf Turkish translation by yavuz kaya and Ýbrahim erken
+    + included clamdoc.pdf Turkish translation by yavuz kaya and ï¿½brahim erken
     + included clamav-mirror-howto.pdf by Luca Gibelli
     + included clamd+daemontools HOWTO by Jesse D. Guardiani
     + included signatures.pdf
diff --git a/configure.ac b/configure.ac
index 289a0b9..7eb8c05 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,7 +20,7 @@ dnl   MA 02110-1301, USA.
 AC_PREREQ([2.59])
 dnl For a release change [devel] to the real version [0.xy]
 dnl also change VERSION below
-AC_INIT([ClamAV], [0.99.2], [http://bugs.clamav.net/], [clamav], [http://www.clamav.net/])
+AC_INIT([ClamAV], [0.99.2.1], [http://bugs.clamav.net/], [clamav], [http://www.clamav.net/])
 
 AH_BOTTOM([#include "platform.h"])
 dnl put configure auxiliary into config
diff --git a/docs/clamdoc.tex b/docs/clamdoc.tex
index 206a1b2..cb996af 100644
--- a/docs/clamdoc.tex
+++ b/docs/clamdoc.tex
@@ -1,6 +1,6 @@
 %  Clam AntiVirus: User Manual
 %
-%  Copyright (C) 2016 Cisco Systems, Inc.
+%  Copyright (C) 2016-2018 Cisco Systems, Inc.
 %  Copyright (C) 2008-2013 Sourcefire, Inc.
 %  Copyright (C) 2002 - 2007 Tomasz Kojm <tkojm*clamav.net>
 %  Version 0.2x corrected by Dennis Leeuw <dleeuw*made-it.com>
@@ -72,7 +72,7 @@
     \vspace{3cm}
     \begin{flushright}
 	\rule[-1ex]{8cm}{3pt}\\
-	\huge Clam AntiVirus 0.99.2\\
+	\huge Clam AntiVirus 0.99.2.1\\
 	\huge \emph{User Manual}\\
     \end{flushright}
 
@@ -84,6 +84,7 @@
     \noindent
     \begin{boxedminipage}[b]{\textwidth}
     ClamAV User Manual,
+87d
     \copyright \  2016 Cisco Systems, Inc.
     Authors: Tomasz Kojm\\
     This document is distributed under the terms of the GNU General
@@ -1740,7 +1741,7 @@ Verification OK.
 	\item Stefano Rizzetto
 	\item Roaring Penguin Software Inc. (\url{http://www.roaringpenguin.com/})
 	\item Luke Rosenthal
-	\item Jenny Säfström (\url{http://PokerListings.com})
+	\item Jenny Sï¿½fstrï¿½m (\url{http://PokerListings.com})
 	\item School of Engineering, University of Pennsylvania (\url{http://www.seas.upenn.edu/})
 	\item Tim Scoff
 	\item Seattle Server (\url{http://www.seattleserver.com/})
diff --git a/m4/reorganization/version.m4 b/m4/reorganization/version.m4
index 4935b45..e3c3dfc 100644
--- a/m4/reorganization/version.m4
+++ b/m4/reorganization/version.m4
@@ -1,6 +1,6 @@
 dnl change this on a release
 dnl VERSION="devel-`date +%Y%m%d`"
-VERSION="0.99.2"
+VERSION="0.99.2.1"
 
 LC_CURRENT=8
 LC_REVISION=1
